docs for introducing breaking changes in local builds api (#408)
expo/eas-cli: CONTRIBUTING.md

1. Update `CHANGELOG.md`.
   - Add the new section with the release version in the title.
   - Leave the `main` section empty (without any changelog entries).

> 

Reviewer: ```suggestion
```

> 1. If `[EAS BUILD API]` tag is present in CHANGELOG

Reviewer: ```suggestion
1. Publish `eas-cli-local-build-plugin` if any of the changelog entries have been tagged with `[EAS BUILD API]`
```

>    - Find version of the package `@expo/eas-build-job` used in the eas-cli.

Reviewer: ```suggestion
   - Open `package.json` and note the version of `@expo/eas-build-job` used in the eas-cli.
```

>    - Find `Publish` commit in [expo/eas-build](https://github.com/expo/eas-build/commit/) for that version of `@expo/eas-build-job`. In most cases it will be in the latest publish commit.

Reviewer: ```suggestion
   - Find the `Publish` commit in [expo/eas-build](https://github.com/expo/eas-build/commit/) for that version. In most cases, it will be the latest `Publish` commit.
```

>    - Find version of the package `eas-cli-local-build-plugin` published in that commit.

Reviewer: ```suggestion
   - Find the `eas-cli-local-build-plugin` version published in that commit.
```

Reviewer: <img width="702" alt="Screenshot 2021-05-14 at 16 27 34" src="https://user-images.githubusercontent.com/5256730/118285148-74763300-b4d1-11eb-979a-0f8eaade02cb.png">
Maybe add this screen to make the instructions easier to follow.

>    - Run `npm dist-tags add eas-cli-local-build-plugin@VERSION latest` with the version from previous step.

Reviewer: ```suggestion
   - Run `npm dist-tags add eas-cli-local-build-plugin@VERSION latest` with the version from the previous step.
```
